tests/api_tests/collections/test_views.py

```diff
@@ -0,0 +1,622 @@
+        # a little better
+        assert_equal(res.status_code, 403)
+
+    def test_return_private_collection_children_list_logged_in_contributor(self):
+        res = self.app.get(self.private_collection_url, auth=self.basic_auth)
+        assert_equal(res.status_code, 200)
+        assert_equal(len(res.json['data']), 1)
+        assert_equal(res.json['data'][0]['id'], self.component._id)
+
+    def test_smart_folders_in_dashboard_children(self):
+        res = self.app.get(self.dashboard_children_url, auth=self.basic_auth)
+        res_json = res.json['data']
+
+        titles = [node['title'] for node in res_json]
+        assert_equal(res.status_code, 200)
+        assert_in('Smart Folder ~amr', titles)
+        assert_in('Smart Folder ~amp', titles)
+
+        Node.remove()
+
+
+class TestCollectionPointersList(ApiTestCase):
+    def setUp(self):
+        ApiTestCase.setUp(self)
+        self.user = UserFactory.build()
+        self.user.set_password('password')
+        self.user.save()
+        self.basic_auth = (self.user.username, 'password')
+        self.collection = FolderFactory(is_public=False, creator=self.user)
+        self.pointer_collection = FolderFactory(is_public=False, creator=self.user)
+        self.collection.add_pointer(self.pointer_collection, auth=Auth(self.user))
+        self.private_url = '/{}collections/{}/pointers/'.format(API_BASE, self.collection._id)
+
+        self.public_folder = FolderFactory(is_public=True, creator=self.user)
+        self.public_pointer_collection = FolderFactory(is_public=True, creator=self.user)
+        self.public_folder.add_pointer(self.public_pointer_collection, auth=Auth(self.user))
+        self.public_url = '/{}collections/{}/pointers/'.format(API_BASE, self.public_folder._id)
+
+        self.user_two = UserFactory.build()
+        self.user_two.set_password('password')
+        self.user_two.save()
+        self.basic_auth_two = (self.user_two.username, 'password')
+
+    def test_return_public_collection_pointers_logged_out(self):
+        res = self.app.get(self.public_url, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_return_public_collection_pointers_logged_in(self):
+        res = self.app.get(self.public_url, auth=self.basic_auth_two)
+        res_json = res.json['data']
+        assert_equal(len(res_json), 1)
+        assert_equal(res.status_code, 200)
+        assert_in(res_json[0]['node_id'], self.public_pointer_collection._id)
+
+    def test_return_private_collection_pointers_logged_out(self):
+        res = self.app.get(self.private_url, expect_errors=True)
+        # This is 403 instead of 401 because basic authentication is only for unit tests and, in order to keep from
+        # presenting a basic authentication dialog box in the front end. We may change this as we understand CAS
+        # a little better
+        assert_equal(res.status_code, 403)
+
```

Comment:
And a test for smart folders returning the proper information in their pointers list as well.